File: android/SalmonTrollingAssistant/app/src/main/java/com/example/salmontrollingassistant/domain/service/CatchLoggingService.kt
package com.example.salmontrollingassistant.domain.service

import com.example.salmontrollingassistant.domain.model.CatchData
import kotlinx.coroutines.flow.Flow

interface CatchLoggingService {
    /**
     * Log a new catch
     */
    suspend fun logCatch(catchData: CatchData): Result<Boolean>
    
    /**
     * Get all catch history
     */
    fun getCatchHistory(): Flow<List<CatchData>>
    
    /**
     * Get catch history for a specific location
     */
    fun getCatchHistoryByLocation(locationId: String): Flow<List<CatchData>>
    
    /**
     * Get catch history for a specific species
     */
    fun getCatchHistoryBySpecies(species: String): Flow<List<CatchData>>
    
    /**
     * Get a specific catch by ID
     */
    suspend fun getCatchById(catchId: String): Result<CatchData>
    
    /**
     * Update an existing catch
     */
    suspend fun updateCatch(catchData: CatchData): Result<Boolean>
    
    /**
     * Delete a catch
     */
    suspend fun deleteCatch(catchId: String): Result<Boolean>
    
    /**
     * Add a photo to a catch
     */
    suspend fun addPhotoCatch(catchId: String, photoUri: String): Result<Boolean>
    
    /**
     * Remove a photo from a catch
     */
    suspend fun removePhotoCatch(catchId: String, photoUri: String): Result<Boolean>
}

File: ios/SalmonTrollingAssistant/SalmonTrollingAssistant/Domain/Services/CatchLoggingService.swift
import Foundation
import Combine

protocol CatchLoggingService {
    /**
     * Log a new catch
     */
    func logCatch(_ catchData: CatchData) async -> Result<Bool, Error>
    
    /**
     * Get all catch history
     */
    var catchHistory: CurrentValueSubject<[CatchData], Never> { get }
    
    /**
     * Get catch history for a specific location
     */
    func getCatchHistoryByLocation(locationId: String) -> AnyPublisher<[CatchData], Never>
    
    /**
     * Get catch history for a specific species
     */
    func getCatchHistoryBySpecies(species: FishSpecies) -> AnyPublisher<[CatchData], Never>
    
    /**
     * Get a specific catch by ID
     */
    func getCatchById(catchId: String) async -> Result<CatchData, Error>
    
    /**
     * Update an existing catch
     */
    func updateCatch(_ catchData: CatchData) async -> Result<Bool, Error>
    
    /**
     * Delete a catch
     */
    func deleteCatch(catchId: String) async -> Result<Bool, Error>
    
    /**
     * Add a photo to a catch
     */
    func addPhotoCatch(catchId: String, photoUrl: String) async -> Result<Bool, Error>
    
    /**
     * Remove a photo from a catch
     */
    func removePhotoCatch(catchId: String, photoUrl: String) async -> Result<Bool, Error>
}